server/src/main/java/org/opensearch/index/RemoteRefreshSegmentPressureSettings.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index;
+
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Settings related to back pressure on account of segments upload failures / lags.
+ */
+public class RemoteRefreshSegmentPressureSettings {
+
+    private static class Defaults {
+        private static final long MIN_SEQ_NO_LAG_LIMIT = 5;
+        private static final double BYTES_LAG_VARIANCE_THRESHOLD = 2.0;
+        private static final double TIME_LAG_VARIANCE_THRESHOLD = 2.0;
```

Comment:
Should we set the defaults to higher number , say 5 to begin with ? 

when is this lag calculated - at the beginning of refresh or at the end ?

Reply:
The name   `TIME_LAG_VARIANCE_THRESHOLD` is confusing , should we  have `UPLOAD_TIME_MULTIPLIER` ?

Reply:
> Should we set the defaults to higher number , say 5 to begin with ?

2 should be good to start with. It allows that the time lag is not going beyond twice the average upload time. Cases like segment merges are handled as the first check that if the lag is below 2, we allow the writes. This is required as we have seen in runs that the uploads can take minutes on account of segment merges. Can you elaborate any case that we have missed to analyse?

> when is this lag calculated - at the beginning of refresh or at the end ?

The lag is calculated with every local refresh or remote refresh segments upload.